kindelia_core/src/hvm.rs

```diff
@@ -287,6 +287,92 @@ pub enum Term {
   Op2 { oper: Oper, val0: Box<Term>, val1: Box<Term> },  // FIXME: refactor `oper` u128 to enum
 }
 
+
+impl Drop for Term {
+  fn drop(&mut self) {
+    /// Verify if `term` has recursive childs (any of its
+    /// `Term`'s properties is not a var or a num)
+    fn term_is_recursive(term: &Term) -> bool {
+      fn term_is_num_or_var(term: &Term) -> bool {
+        matches!(term, &Term::Num { .. }) ||
```

Comment:
Why use `matches!` twice? Isn't it clearer to just do
```rs
match term {
  Term::Num { .. } || Term::Var { .. } => true,
  _ => false
}
```

Reply:
Done in 5942dd0